### PR TITLE
OA-293

### PR DESCRIPTION
**Pull request by kathy-nevola**
# Title: Support Flex wide data
**Problem:** Write a description of your problem here.

**Solution:** Write a short summary of your solution.

**Key Features:**

* Key feature 1
* Key feature 2
* ...

## Checklist
- [ ] Make sure you are requesting to **pull a feature/bugfix branch** (right side). This should not be main or develop.
- [ ] Make sure you are make a pull request against either **main or develop** (left side). (Requesting to main should be reserved for bugfixes and new releases)
- [ ] Add or update unit tests (if applicable)
- [ ] Check your code with any unit tests (Run devtools::check() locally)
- [ ] Add neccessary documentation (if applicable)

## Type of changes

What type of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue) (link the issue on the right)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code style update (formatting, renaming)
- [ ] Documentation Update 
- [ ] Other (explain)

## Further comments
If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, any questions you have, etc...

Consider linking any issues (#issue-number ) or adding @mentions to ensure specific people see it.

### CODE REVIEW

File: OlinkAnalyze/R/read_flex.R [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+                  dplyr::matches("Plat_LOD"),
+                  dplyr::matches("LLOQ"),
+                  dplyr::matches("ULOQ"),
+                  dplyr::matches("NPX"),
+                  dplyr::matches("Quantified_value"),
+                  dplyr::matches("Unit"),
+                  dplyr::matches("Assay_Warning"),
+                  dplyr::matches("Normalization"),
+                  dplyr::matches("*Inc Ctrl*"),
+                  dplyr::matches("*Det Ctrl*"))
```
Comment (AskPascal) [inline OlinkAnalyze/R/read_flex.R:172]: Maybe could be simplified with `tidyselect::any_of()`?

Also, if im not mistaken *Inc Ctrl* and *Det Ctrl* are never set with current implemenetation...

File: OlinkAnalyze/R/Read_NPX_data.R
Comment (MasoumehSheikhi) [inline OlinkAnalyze/R/Read_NPX_data.R:1]: General suggestion, not specifically about the changes in this commit, optional
- Adding suppressWarnings() around as.numeric() to suppress the NA introduced warning messages 
- In Read_NPX_data.R, replacing all %>% with |> to unify the pipes

File: OlinkAnalyze/R/Read_NPX_data.R [hunk 2]
```diff
@@ -250,62 +250,102 @@ read_NPX_explore <- function(filename) {
...
+                                   "Olink")))
+  wide_or_long <- ifelse(long_form, "long", "wide")
+  message(paste(platform, "data in", wide_or_long, "form detected.", sep = " "))
+  
+  
+  # Long form
+  if (long_form){
     out <- readxl::read_excel(path = filename,
-                       col_names = T) %>%
+                              col_names = T) %>%
```
Comment (MasoumehSheikhi) [inline OlinkAnalyze/R/Read_NPX_data.R:348]: use TRUE instead of T

File: OlinkAnalyze/R/Read_NPX_data.R [hunk 3]
```diff
@@ -337,76 +379,58 @@ read_NPX_target <- function(filename) {
...
-                    dplyr::matches("Assay_Warning"),
-                    dplyr::matches("Normalization"),
-                    dplyr::matches("*Inc Ctrl*"),
-                    dplyr::matches("*Det Ctrl*"))
-    is_npx_data <- ifelse(any(names(out) %in% "NPX"), TRUE, FALSE)
+    } else if (isFlex){
+    out <- read_flex(filename)
   } else {
-    # Load initial meta data (the first rows of the wide file)
+    target_type <- ifelse(isTarget48, "48", "96")
```
Comment (MasoumehSheikhi) [inline OlinkAnalyze/R/Read_NPX_data.R:385]: local variable target_type assigned but not used

File: OlinkAnalyze/R/Read_NPX_data.R [hunk 4]
```diff
@@ -454,78 +478,78 @@ read_NPX_target <- function(filename) {
     assay_name_list <- list()
     panel_list_long <- list()
     deviations_list <- list()
```
Comment (MasoumehSheikhi) [inline OlinkAnalyze/R/Read_NPX_data.R:480]: local variable deviations_list assigned but not used

File: OlinkAnalyze/R/Read_NPX_data.R
Comment (kathy-nevola) [reply OlinkAnalyze/R/Read_NPX_data.R:1]: resolved pipes but not suppressWarnings()

**Comment by kathy-nevola** [comment]
@MasoumehSheikhi had to roll back |> due to lack of support for . objects.

**Pull request merged**